src/axom/primal/geometry/Polyhedron.hpp

```diff
@@ -669,37 +669,42 @@ class Polyhedron
   }
 
   /*!
- * \brief Creates a Polyhedron from a given Hexahedron's vertices.
- *
- * \param [in] hex The hexahedron
- * \param [in] checkSign If true (default is false), checks if the
- *             signed volume of the Polyhedron is positive. If signed volume
- *             is negative, order of some vertices will be swapped.
- *
- * \return A Polyhedron with the Hexahedron's vertices and added
- *         vertex neighbors
- *
- * \note The Hexahedron is assumed to have a specific vertex order:
- * \verbatim
- *
- *          4--------7          +z
- *         /|       /|               +y
- *        / |      / |           ^  >
- *       5--------6  |           | /
- *       |  0-----|--3           |/
- *       | /      | /            -----> +x
- *       |/       |/
- *       1--------2
- *
- * \endverbatim
- *
- *       The Polyhedron's vertex neighbors are created assuming this vertex
- *       ordering.
- *
- * \note checkSign flag does not guarantee the Polyhedron's vertex order
- *       will be valid. It is the responsiblity of the caller to pass
- *       a Hexahedron with a valid vertex order.
- */
+   * \brief Creates a Polyhedron from a given Hexahedron's vertices.
+   *
+   * \param [in] hex The hexahedron
+   * \param [in] checkSign If true (default is false), checks if the
+   *             signed volume of the Polyhedron is positive. If signed volume
+   *             is negative, order of some vertices will be swapped to try to
+   *             obtain a positive volume for the Polyhedron. Otherwise, if the
+   *             signed volume is negative and the vertices are not swapped,
+   *             the returned Polyhedron will have a non-positive volume.
+   *
+   * \return A Polyhedron with the Hexahedron's vertices and added
+   *         vertex neighbors
+   *
+   * \note The Hexahedron is assumed to have a specific vertex order:
+   * \verbatim
+   *
+   *          4--------7          +z
+   *         /|       /|               +y
+   *        / |      / |           ^  >
+   *       5--------6  |           | /
+   *       |  0-----|--3           |/
+   *       | /      | /            -----> +x
+   *       |/       |/
+   *       1--------2
+   *
+   * \endverbatim
+   *
+   *       The Polyhedron's vertex neighbors are created assuming this vertex
+   *       ordering.
+   *
+   * \warning checkSign flag does not guarantee the Polyhedron's vertex order
```

Comment:
I don't understand. Why not?

Reply:
> I don't understand. Why not?

In the case of a unit cube/ hexahedron for example, an invalid ordering can result in a volume less than 1 due to overlapping faces/edges, inconsistent winding, etc. 

`checkSign` makes sure the volume is positive, but it doesn't protect against the user passing an odd ordering like so:
```
   *          6--------2          +z
   *         /|       /|               +y
   *        / |      / |           ^  >
   *       3--------7  |           | /
   *       |  0-----|--4           |/
   *       | /      | /            -----> +x
   *       |/       |/
   *       5--------1
   *
```